src/core/callstack.c

```diff
@@ -631,8 +634,158 @@ static void handle_bind_control(MVMThreadContext *tc, MVMCallStackBindControl *c
     ice->run_dispatch(tc, ice_ptr, ice, id, callsite, args_map, flag_ptr,
             control_record->sf, 0);
 }
+
+MVM_STATIC_INLINE void cleanup_region_record(MVMThreadContext *tc) {
+    tc->stack_current_region->alloc = (char *)tc->stack_top;
+    tc->stack_current_region = tc->stack_current_region->prev;
+    tc->stack_top = tc->stack_top->prev;
+}
+MVM_STATIC_INLINE void move_to_next_record(MVMThreadContext *tc) {
+    tc->stack_current_region->alloc = (char *)tc->stack_top;
+    tc->stack_top = tc->stack_top->prev;
+}
+MVM_STATIC_INLINE void cleanup_dispatch_recorded_record(MVMThreadContext *tc) {
+    MVMCallStackDispatchRecord *disp_record =
+        (MVMCallStackDispatchRecord *)tc->stack_top;
+    if (disp_record->resumption_state.disp)
+        MVM_disp_resume_destroy_resumption_state(tc, &(disp_record->resumption_state));
+    if (disp_record->produced_dp && !disp_record->produced_dp_installed)
+        MVM_disp_program_destroy(tc, disp_record->produced_dp);
+    if (disp_record->temps)
+        MVM_free(disp_record->temps);
+    tc->stack_current_region->alloc = (char *)tc->stack_top;
+    tc->stack_top = tc->stack_top->prev;
```

Comment:
And since we're factoring out to static inlines, this could also call `move_to_prev_record`

Reply:
(And the same comment applies in a few other places too)